public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -353,6 +353,15 @@ export class DashboardModel implements TimeModel {
 
         if (variable.type === 'adhoc') {
           variableSaveModel.filters = original.filters;
+        } else if (variable.type === 'custom') {
+          // Even if we're not updating the currently selected variable in the
+          // dashboard model, we want to save the options
+          variableSaveModel.current = original.current;
+          const originalOptions: VariableOption[] = original.options;
+          const oldSelectedOptions = new Set(originalOptions.filter((o) => o.selected).map((o) => o.value));
+          for (const option of variableSaveModel.options) {
+            option.selected = oldSelectedOptions.has(option.value);
+          }
```

Comment:
Do we have a bug report for this? Can only impact reporting right (the only part of grafana that uses the saved options) ?

Reply:
No bug report, but there's an escalation: https://github.com/grafana/support-escalations/issues/10027

Unfortunately I think it affects more than reporting, at least in the old architecture, as the options picker's options are taken from the variable's options param rather than derived from the query string 
https://github.com/grafana/grafana/blob/917cbce448e7e89d69b1b70c59cedaa8d6ffc956/public/app/features/variables/pickers/OptionsPicker/reducer.ts#L131-L151

Reply:
> Unfortunately I think it affects more than reporting, at least in the old architecture, as the options picker's options are taken from the variable's options param rather than derived from the query string

@kaydelaney sure but the options are re-built from the query string every time on dashboard load, so still a bit confused here

Reply:
@kaydelaney looking at that escalation I do not understand really how this fixes it. 

Why does setting a value in URL cause 1 of the options defined in the query string to go mission in the options picker?

Reply:
@kaydelaney so tracked it down, Custom variables don' have a refresh property so this function returns never here:
https://github.com/grafana/grafana/blob/main/public/app/features/variables/utils.ts#L159 

Called from here: https://github.com/grafana/grafana/blob/main/public/app/features/variables/state/actions.ts#L403

That is why this bug does happen for Query variables

Reply:
I see. I suppose the simplest fix would be to add the refresh prop the custom vars?